src/mode_RPiHQ_mean.cpp

```diff
@@ -143,202 +57,178 @@ void RPiHQcalcMean(const char* fileName, int asiExposure_us, double asiGain, ras
+			currentModeMeanSetting.ExposureLevel += ExposureChange;
 		}
-		else if (deltaGain < -2.0) {
-			currentRaspistillSetting.analoggain -= 2.0;
+	}
+	if (mean > (currentModeMeanSetting.mean_value + currentModeMeanSetting.mean_threshold))  {
+		if (ExposureTime_s <= 1 / US_IN_SEC) { // untere Grenze durch shuttertime
+			Log(2, "ExposureTime_s too low - stop !\n");
 		}
 		else {
-			currentRaspistillSetting.analoggain = newGain;
+			currentModeMeanSetting.ExposureLevel -= ExposureChange;
 		}
-		// min=1µs, max asiExposure
-		ExposureTime_s = std::min(asiExposure_us/1000000.0,std::max(0.000001, pow(2.0, double(currentModeMeanSetting.ExposureLevel)/pow(currentModeMeanSetting.shuttersteps,2.0)) / currentRaspistillSetting.analoggain));
+	}
 
-		//#############################################################################################################
-		// prepare for the next measurement
-		if (currentModeMeanSetting.quickstart > 0) {
-			currentModeMeanSetting.quickstart--;
-		}
-		// Exposure gilt fuer die naechste Messung
-		MeanCnt++;
-		exp_history[MeanCnt % currentModeMeanSetting.historySize] = currentModeMeanSetting.ExposureLevel;
+	// check limits of exposurelevel 
+	currentModeMeanSetting.ExposureLevel = std::max(std::min((int)currentModeMeanSetting.ExposureLevel, (int)currentModeMeanSetting.ExposureLevelMax), (int)currentModeMeanSetting.ExposureLevelMin);
 
-		currentRaspistillSetting.shutter_us = ExposureTime_s * 1000 * 1000;
-		printf("Mean: %1.4f (%1.4f) Exposure level:%d (%d) Exposure time:%1.8f analoggain:%1.2f\n", mean, mean_diff, currentModeMeanSetting.ExposureLevel, currentModeMeanSetting.ExposureLevel-exp_history[idx], ExposureTime_s, currentRaspistillSetting.analoggain);
+	// fastforward ?
+	if ((currentModeMeanSetting.ExposureLevel == (int)currentModeMeanSetting.ExposureLevelMax) || (currentModeMeanSetting.ExposureLevel == (int)currentModeMeanSetting.ExposureLevelMin)) {
+		fastforward = true;
+		printf("FF aktiviert\n");
+	}
+	if ((abs(mean_history[idx] - currentModeMeanSetting.mean_value) < currentModeMeanSetting.mean_threshold) &&
+		(abs(mean_history[idxN1] - currentModeMeanSetting.mean_value) < currentModeMeanSetting.mean_threshold)) {
+		fastforward = false;
+		printf("FF deaktiviert\n");
+	}
+		
+	//#############################################################################################################
+	// calculate gain und exposuretime
+	double newGain = std::min(asiGain, std::max(1.0, pow(2.0, double(currentModeMeanSetting.ExposureLevel)/pow(currentModeMeanSetting.shuttersteps,2.0)) / (asiExposure_us/US_IN_SEC))); 
+	double deltaGain = newGain - currentRaspistillSetting.analoggain; 
+	if (deltaGain > 2.0) {
+		currentRaspistillSetting.analoggain += 2.0;
+	}
+	else if (deltaGain < -2.0) {
+		currentRaspistillSetting.analoggain -= 2.0;
 	}
-}
\ No newline at end of file
+	else {
+		currentRaspistillSetting.analoggain = newGain;
+	}
+	// min=1Âµs, max asiExposure
```

Comment:
typo?: 1µs or 1us

Reply:
In the ZWO version I gave up trying to make it look like "µ" and now just use "us".  "µ" appeared fine in some editors and as "Âµs" in others.

I did a commit to fix this.